### PR TITLE
[on hold] Rewrite tests and examples to use accessors

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR originated as part of #1249 and rewrites all alpaka examples and tests to use the new accessors facility.

### CODE REVIEW

**Comment by SimeonEhrig** [comment]
I see, some examples are not ported to accessors. I'm not sure, if this was intention or only missing, because some examples was added in the mean time. Nevertheless, I for it, that the example `randomCells2D` should not be ported, because it demonstrate the usage of pointers as kernel arguments and the usage of the pitch.

File: test/integ/axpy/src/axpy.cpp [hunk 1]
```diff
@@ -166,10 +166,9 @@ TEMPLATE_LIST_TEST_CASE("axpy", "[axpy]", TestAccs)
     auto const taskKernel = alpaka::createTaskKernel<Acc>(
         workDiv,
         kernel,
-        numElements,
         alpha,
-        alpaka::getPtrNative(memBufAccX),
-        alpaka::getPtrNative(memBufAccY));
+        alpaka::experimental::readAccess(memBufAccX),
```
Comment (j-stephan) [inline test/integ/axpy/src/axpy.cpp:170]: clang complains about this when in C++17 mode:

```
/usr/lib64/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/type_traits:2933:5: error: no type named 'type' in 'std::invoke_result<AxpyKernel, const alpaka::AccCpuSerial<std::integral_constant<unsigned long, 1>, unsigned long> &, const unsigned long &, const float &, const alpaka::experimental::Accessor<float *, float, unsigned long, 1, alpaka::experimental::ReadAccess> &, const alpaka::experimental::Accessor<float *, float, unsigned long, 1, alpaka::experimental::ReadWriteAccess> &>'
    using invoke_result_t = typename invoke_result<_Fn, _Args...>::type;
    ^~~~~
```

The accessors created here are passed by reference to the lower layers but the kernel (correctly) expects them to be passed by value.

Comment (bernhardmgruber) [reply test/integ/axpy/src/axpy.cpp:170]: I can try to have a look later.

Comment (bernhardmgruber) [reply test/integ/axpy/src/axpy.cpp:170]: I can build all tests and examples with clang-12 locally now.

Comment (j-stephan) [reply test/integ/axpy/src/axpy.cpp:170]: DPCPP appears to be based on clang-13. Maybe something changed between clang versions?

Comment (bernhardmgruber) [reply test/integ/axpy/src/axpy.cpp:170]: I guess we need to get #1485 in then.

Comment (j-stephan) [reply test/integ/axpy/src/axpy.cpp:170]: I think I found the culprit:

https://github.com/alpaka-group/alpaka/blob/f9b1b7f7b37954288ad08ad905fa577ee1776c42/include/alpaka/kernel/Traits.hpp#L208-L218

Note that all arguments are transformed into a `const` reference during the check.

No idea why this works for other compilers, though.

Comment (j-stephan) [reply test/integ/axpy/src/axpy.cpp:170]: Never mind. It turns out that I'm stupid and forgot to delete the `numElements` parameter from the kernel instantiation call. Everything works as expected now.

File: test/integ/axpy/src/axpy.cpp [hunk 2]
```diff
@@ -30,25 +30,25 @@ class AxpyKernel
...
+        alpaka::experimental::Accessor<TMemoryHandleY, TElem, TIdx, 1, alpaka::experimental::ReadWriteAccess> const Y)
+        const -> void
     {
         static_assert(alpaka::Dim<TAcc>::value == 1, "The AxpyKernel expects 1-dimensional indices!");
 
         auto const gridThreadIdx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u];
         auto const threadElemExtent = alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u];
         auto const threadFirstElemIdx = gridThreadIdx * threadElemExtent;
 
+        auto const numElements = X.extents[0];
```
Comment (j-stephan) [inline test/integ/axpy/src/axpy.cpp:51]: The `X.extents[0]` isn't part of the `Accessor` interface (it only belongs to `ViewAccessor`).

Comment (j-stephan) [reply test/integ/axpy/src/axpy.cpp:51]: I wonder whether this should be obtainable through `getExtent`, too.

Comment (bernhardmgruber) [reply test/integ/axpy/src/axpy.cpp:51]: IIRC, we don't have any other accessor than a ViewAccessor at the moment. And yes, we could think about accessors implementing `getExtent`. For now I went with the easier solution.

Comment (j-stephan) [reply test/integ/axpy/src/axpy.cpp:51]: Well, there are my SYCL accessors ;-)

**Comment by bernhardmgruber** [comment]
I would really like to abandon this work in favor of `mdspan`. Let's see how far we get with the latter.

**Comment by j-stephan** [comment]
Since we now have `mdspan` and have decided to abandon accessors we can close this PR, right?

**Comment by bernhardmgruber** [comment]
> Since we now have `mdspan` and have decided to abandon accessors we can close this PR, right?

In principle, yes. I will just have a look at the PR again whether there is something else that's useful.

**Comment by bernhardmgruber** [comment]
Ok, the fixes for the `bufferCopy` example where done with `mdspan` as well. Looks like we can close it :)

**Pull request closed without merging**